assets/kube-state-metrics/deployment.yaml

```diff
@@ -57,7 +57,24 @@ spec:
           ^kube_pod_completion_time$,
           ^kube_pod_status_scheduled$
         image: registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.12.0
+        livenessProbe:
+          exec:
+            command:
+            - sh
+            - -c
+            - if [ -x "$(command -v curl)" ]; then exec curl -s -I -f http://localhost:8081/livez; elif [ -x "$(command -v wget)" ]; then exec wget --quiet --tries=1 --spider http://localhost:8081/livez; else exit 1; fi
```

Comment:
/livez returns 200 OK currently because the HTTP server returns the home URL. I'd rather have the /livez endpoint implemented before having the change here.

Reply:
I've raised a PR for `livez` here: https://github.com/kubernetes/kube-state-metrics/pull/2418.